lib/WebService/Braintree/Customer.pm

```diff
@@ -68,6 +88,82 @@ sub gateway {
     return WebService::Braintree->configuration->gateway;
 }
 
+sub BUILD {
+    my ($self, $attributes) = @_;
+    my $sub_objects = {
+        addresses => "WebService::Braintree::Address",
+        credit_cards => "WebService::Braintree::CreditCard",
+        paypal_accounts => "WebService::Braintree::PayPalAccount",
+    };
+
+    $self->setup_sub_objects($self, $attributes, $sub_objects);
+    $self->set_attributes_from_hash($self, $attributes);
+}
+
+=head1 OBJECT METHODS
+
+In addition to the methods provided by the keys returned from Braintree, this
+class provides the following methods:
+
+=head2 payment_types()
+
+This returns a list of all the payment types supported by this class.
+
+=cut
+
+sub payment_types {
+    return qw(credit_cards paypal_accounts);
+}
+
+=head2 payment_methods()
+
+This returns an arrayref of all available payment methods across all types.
+
+=cut
+
+#sub __payment_methods {
```

Comment:
I added that version as a better-written version of `payment_methods()`. Since they're inspectably identical, I removed the original. Please let me know if I screwed it up.